index.js

```diff
@@ -180,72 +180,92 @@
   //  Value :: Location
   var Value = 'Value';
 
-  //  _funcPath :: (Boolean, Array String, a) -> Nullable Function
-  function _funcPath(allowInheritedProps, path, _x) {
-    var x = _x;
-    for (var idx = 0; idx < path.length; idx += 1) {
-      var k = path[idx];
-      if (x == null || !(allowInheritedProps || has(k, x))) return null;
-      x = x[k];
-    }
-    return typeof x === 'function' ? x : null;
-  }
+  //  prefix :: String
+  var prefix = 'fantasy-land/';
 
   //  funcPath :: (Array String, a) -> Nullable Function
-  function funcPath(path, x) {
-    return _funcPath(true, path, x);
+  function funcPath(path, value) {
+    var x = value;
+    for (var idx = 0; idx < path.length && x != null; idx += 1) {
+      x = x[path[idx]];
+    }
+    return typeof x === 'function' ? x : null;
   }
 
   //  implPath :: Array String -> Nullable Function
   function implPath(path) {
-    return _funcPath(false, path, implementations);
+    var x = implementations;
+    for (var idx = 0; idx < path.length && has(path[idx], x); idx += 1) {
+      x = x[path[idx]];
+    }
+    return typeof x === 'function' ? x : null;
   }
 
   //  functionName :: Function -> String
-  var functionName = 'name' in function f() {} ?
+  var functionName = 'name' in Function ?
     function functionName(f) { return f.name; } :
     /* istanbul ignore next */
     function functionName(f) {
       var match = /function (\w*)/.exec(f);
       return match == null ? '' : match[1];
     };
 
+  //  getFunction :: String -> TypeRep -> Nullable Function
+  function getFunction(_name) {
+    var name = prefix + _name;
+    return function(typeRep) {
+      var f = funcPath([name], typeRep);
+      return f == null && typeof typeRep === 'function' ?
+        implPath([functionName(typeRep), name]) :
+        f;
+    };
+  }
+
+  //  getBoundMethod :: String -> a -> Nullable (a ~> b)
```

Comment:
I think since we always represent a nullary *methods* as `a ~> () -> b`, this signature is valid. Because `() -> b` is a member of `b`? We could probably even make it more precise by changing it to `a ~> Function`.

Come to think of it, the `~>` is very strange. I'm currently using it like a "return" arrow, but it's not quite that. It shouldn't have a return type unless we allow `a ~> b` to represent a nullary method. It's currently completely meaningless if not followed by an argument list. It's actually more like the `~> ()`-operator. So in other words: I'm using a made-up operator, and I will cease to do so immediately.

---

### Some rambling

Why don't we allow `a ~> b`? I think we can still represent everything we want with it:

```hs
f :: a ~> b -- nullary method returning anything
g :: a ~> (b, c) -> d -- binary method returning anything
h :: a ~> ((b, c) -> d) -- nullary method on A returning a binary function
i :: a ~> Function -- a method of unknown arity
j :: a ~> (Function) -- a method returning a function of unknown arity
bind :: a -> Function -> (a ~> Function) -- binary function returning a method with unknown arity
```

`h` and `j` are very strange, but I think we already have this problem in the way we represent ramda-style curried functions vs actual curried functions:

```hs
f :: a -> b -> c -- we can call f(a, b)
g :: a -> (b -> c) -- we must call g(a)(b)
```